string.c

```diff
@@ -8022,37 +8022,50 @@ rb_str_tr_bang(VALUE str, VALUE src, VALUE repl)
 
 /*
  *  call-seq:
- *     str.tr(from_str, to_str)   => new_str
+ *    tr(replaceables, replacements) -> new_string
+ *
+ *  Returns a copy of +self+ with each character specified by string +replaceables+
+ *  translated to the corresponding character in string +replacements+.
+ *  The correspondence is _positional_:
+ *
+ *  - Each occurrence of the first character in +replaceables+
+ *    is translated to the first character in +replacements+.
+ *  - Each occurrence of the second character in +replaceables+
+ *    is translated to the second character in +replacements+.
+ *  - And so on.
+ *
+ *  Example:
+ *
+ *    'hello'.tr('el', 'ip') #=> "hippo"
+ *
+ *  If +replacements+ is shorter than +replaceables+,
+ *  it is implicitly padded with its own last character:
+ *
+ *    'hello'.tr('aeiou', '-')   # => "h-ll-"
+ *    'hello'.tr('aeiou', 'AA-') # => "hAll-"
  *
- *  Returns a copy of +str+ with the characters in +from_str+ replaced by the
- *  corresponding characters in +to_str+.  If +to_str+ is shorter than
- *  +from_str+, it is padded with its last character in order to maintain the
- *  correspondence.
+ *  Three characters may get special treatment:
  *
- *     "hello".tr('el', 'ip')      #=> "hippo"
- *     "hello".tr('aeiou', '*')    #=> "h*ll*"
- *     "hello".tr('aeiou', 'AA*')  #=> "hAll*"
+ *  - A circumflex (<tt>'^'</tt>) in +replaceables+
+ *    operates as a "not" operator for the following characters in +replaceables+:
  *
- *  Both strings may use the <code>c1-c2</code> notation to denote ranges of
- *  characters, and +from_str+ may start with a <code>^</code>, which denotes
- *  all characters except those listed.
+ *      'hello'.tr('^aeiou', '-') # => "-e--o"
+ *      'hello'.tr('ae^iou', '-') # => "h-ll-"
```

Comment:
I'm a bit confused as to what this example is suggesting. Is it suggesting that a `^` is not treated as a special character unless it's the first character in the `replaceables` string? If that's the case, I think that it should be made more obvious. E.g.

```suggestion
 *      '^hello'.tr('ae^iou', '-') # => "-h-ll-"
```

Reply:
Fixed.